main/src/cgeo/geocaching/models/CalcState.java

```diff
@@ -0,0 +1,44 @@
+package cgeo.geocaching.models;
+
+import java.io.Serializable;
+import java.util.List;
+
+import cgeo.geocaching.settings.Settings;
+import cgeo.geocaching.ui.CalculateButton;
+import cgeo.geocaching.ui.CalculatorVariable;
+
+/**
+ * This class is designed to capture the current state of the coordinate calculator such that it can be preserver for latter use.
+ **
+ * All the relevant information is in a serializable form such that it can be stored as a bundle in waypoint's 'ContentValues'.
+ **/
+
+public class CalcState implements Serializable {
+    public Settings.CoordInputFormatEnum format;
+    public String plainLat, plainLon;
+    public char latHemisphere, lonHemisphere;
+    public List<CalculateButton.ButtonData> buttons;
+    public List<CalculatorVariable.VariableData> equations, freeVariables;
+    public String notes; // Note, we have to use a String rather than an Editable as Editable's can't be serialized
+
+    public CalcState(Settings.CoordInputFormatEnum theFormat,
```

Comment:
As you may have noticed, in addition to making all parameters `final` to ensure that they are not used as a reassignable local variable within the method (or the constructor), we also use the same parameter names as the field names in constructors.

This should look like

``` java
   public CalcState(final Settings.CoordInputFormatEnum format, …) {
      this.format = format;
      …
   }
```

This clearly reflects that the parameter is a simple initializer for the field using the same name.